streams-contrib/streams-provider-twitter/src/main/java/org/apache/streams/twitter/provider/TwitterFollowersProviderTask.java

```diff
@@ -0,0 +1,181 @@
+
+    protected TwitterFollowingProvider provider;
+    protected Twitter client;
+    protected Long id;
+    protected String screenName;
+
+    public TwitterFollowersProviderTask(TwitterFollowingProvider provider, Twitter twitter, Long id) {
+        this.provider = provider;
+        this.client = twitter;
+        this.id = id;
+    }
+
+    public TwitterFollowersProviderTask(TwitterFollowingProvider provider, Twitter twitter, String screenName) {
+        this.provider = provider;
+        this.client = twitter;
+        this.screenName = screenName;
+    }
+
+
+    @Override
+    public void run() {
+
+        if( id != null )
+            getFollowers(id);
+        if( screenName != null)
+            getFollowers(screenName);
+
+        LOGGER.info(id != null ? id.toString() : screenName + " Thread Finished");
+
+    }
+
+    protected void getFollowers(Long id) {
+
+        int keepTrying = 0;
+
+        long curser = -1;
+
+        do
+        {
+            try
+            {
+                twitter4j.User followee4j;
+                String followeeJson;
+                try {
+                    followee4j = client.users().showUser(id);
+                    followeeJson = TwitterObjectFactory.getRawJSON(followee4j);
+                } catch (TwitterException e) {
+                    LOGGER.error("Failure looking up " + id);
+                    break;
+                }
+
+                PagableResponseList<twitter4j.User> followerList = client.friendsFollowers().getFollowersList(id.longValue(), curser);
+
+                for (twitter4j.User follower4j : followerList) {
+
+                    String followerJson = TwitterObjectFactory.getRawJSON(follower4j);
+
+                    try {
+                        Follow follow = new Follow()
+                                .withFollowee(mapper.readValue(followeeJson, User.class))
+                                .withFollower(mapper.readValue(followerJson, User.class));
+
+                        ComponentUtils.offerUntilSuccess(new StreamsDatum(follow), provider.providerQueue);
+                    } catch (JsonParseException e) {
+                        LOGGER.warn(e.getMessage());
+                    } catch (JsonMappingException e) {
+                        LOGGER.warn(e.getMessage());
+                    } catch (IOException e) {
+                        LOGGER.warn(e.getMessage());
+                    }
+                }
+                curser = followerList.getNextCursor();
+            }
+            catch(TwitterException twitterException) {
+                keepTrying += TwitterErrorHandler.handleTwitterError(client, twitterException);
+            }
+            catch(Exception e) {
+                keepTrying += TwitterErrorHandler.handleTwitterError(client, e);
+            }
+        } while (curser != 0 && keepTrying < 10);
+    }
+
+    protected void getFollowers(String screenName) {
```

Comment:
Since showUser() is called, getFollowers(Long id) could be the primary implementation with getFollowers(String screenName) calling it after the id is available from the user lookup. This would cut out a bit of code.